src/Datum.ts

```diff
@@ -275,18 +280,81 @@ export function getOrd<A>(O: Ord<A>): Ord<Datum<A>> {
  * `Apply` semigroup
  *
  * @since 2.0.0
+ * 
+ * Note: Semantics changed in 4.0.0 to match new Apply instance
  */
-export const getApplySemigroup = <A>(S: Semigroup<A>): Semigroup<Datum<A>> =>
-  getSemigroup(S);
+export const getApplySemigroup = <A>(S: semigroup.Semigroup<A>): semigroup.Semigroup<Datum<A>> => ({
+  // TODO: replace with apply.getApplySemigroup if we bump the min supported version of fp-ts >= 2.10
+  // Or just remove in place of that factory instance
+  concat: (first: Datum<A>, second: Datum<A>) =>
+    Apply.ap(
+      Apply.map(first, (x: A) => (y: A) => S.concat(x, y)),
+      second
+    )
+})
 
 /**
  * @since 2.0.0
+ * 
+ * @deprecated Does not match getApplySemigroup. Use getMonoid for the same behavior under a new name.
  */
-export const getApplyMonoid = <A>(M: Monoid<A>): Monoid<Datum<A>> => ({
-  ...getApplySemigroup(M),
+export const getApplyMonoid = <A>(M: Monoid<A>): Monoid<Datum<A>> => getMonoid(M)
```

Comment:
There actually isn't a true ApplyMonoid given the new Apply behavior. The old monoid behavior is still included, below. We could also just drop this function now if you want.

Reply:
Leaving this as is for now with a deprecation tag and a note that it will be removed in 4.0 would be best. As for getMonoid and getProgressSemigroup I think those would be best implemented as the default getMonoid and getSemigroup functions in 4.0. What do you think?